plugin/trino-base-jdbc/src/main/java/io/trino/plugin/jdbc/expression/GenericRewrite.java

```diff
@@ -0,0 +1,94 @@
+/*
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package io.trino.plugin.jdbc.expression;
+
+import io.trino.matching.Captures;
+import io.trino.plugin.base.expression.ConnectorExpressionRule;
+import io.trino.spi.expression.ConnectorExpression;
+
+import java.util.Optional;
+import java.util.regex.Matcher;
+import java.util.regex.Pattern;
+
+import static java.lang.String.format;
+import static java.util.Objects.requireNonNull;
+import static java.util.regex.Matcher.quoteReplacement;
+
+public class GenericRewrite
+        implements ConnectorExpressionRule<ConnectorExpression, String>
+{
+    private static final Pattern REWRITE_TOKENS = Pattern.compile("(?<![a-zA-Z0-9_$])[a-zA-Z_$][a-zA-Z0-9_$]*(?![a-zA-Z0-9_$])");
```

Comment:
Is it possible to do this in more robust manner? This is too magical for my taste. It's extracting tokens from the rewritePattern.

Reply:
If i call it "REWRITE_WORDS" would it address your concerns?

Note that rewrite target is a foreign language that we don't speak. We cannot have any kind of grammar for it.

Reply:
REWRITE_EXPRESSION_TOKENS would be fine.

Reply:
it's not really expression.

Reply:
Will add a comment
```
// Matches words in the `rewritePattern`.
```